test/test_linalg.py

```diff
@@ -5689,28 +5689,76 @@ def run_test(matsize, batchdims, mat_chars):
+        def run_test(shape, batch, upper, contiguous):
+            A = random_hermitian_pd_matrix(shape, *batch, dtype=dtype, device=device)
+            if A.numel() > 0 and not contiguous:
+                A = A.transpose(-2, -1)
+                self.assertFalse(A.is_contiguous())
+            L = torch.linalg.cholesky(A)
+            expected_inverse = torch.inverse(A)
+            L = L.conj().transpose(-2, -1) if upper else L
+            actual_inverse = torch.cholesky_inverse(L, upper)
+            self.assertEqual(actual_inverse, expected_inverse)
 
-        # default case
-        chol = torch.cholesky(a)
-        inv1 = torch.cholesky_inverse(chol, False)
-        self.assertLessEqual(inv0.dist(inv1), 1e-12)
+        shapes = (0, 3, 5)
+        batches = ((), (0,), (3, ), (2, 2))
+        for shape, batch, upper, contiguous in list(itertools.product(shapes, batches, (True, False), (True, False))):
+            run_test(shape, batch, upper, contiguous)
 
-        # upper Triangular Test
-        chol = torch.cholesky(a, True)
-        inv1 = torch.cholesky_inverse(chol, True)
-        self.assertLessEqual(inv0.dist(inv1), 1e-12)
+        # check the out= variant
+        A = random_hermitian_pd_matrix(3, 2, dtype=dtype, device=device)
+        L = torch.linalg.cholesky(A)
 
-        # lower Triangular Test
-        chol = torch.cholesky(a, False)
-        inv1 = torch.cholesky_inverse(chol, False)
-        self.assertLessEqual(inv0.dist(inv1), 1e-12)
+        # batched column major out (memory is used in computations directly)
+        out = torch.empty_like(A)
+        out_t = out.transpose(-2, -1).clone(memory_format=torch.contiguous_format)
+        out = out_t.transpose(-2, -1)
+        ans = torch.cholesky_inverse(L, out=out)
+        self.assertEqual(ans, out)
+        expected = torch.inverse(A)
+        self.assertEqual(expected, out)
+        # batched row major out (temporary tensor is allocated internally and result is copied to out)
+        out = torch.empty_like(A)
+        ans = torch.cholesky_inverse(L, out=out)
+        self.assertEqual(ans, out)
+        expected = torch.inverse(A)
+        self.assertEqual(expected, out)
+
+    @skipCUDAIfNoMagma
+    @skipCPUIfNoLapack
+    @dtypes(torch.float32, torch.float64, torch.complex64, torch.complex128)
+    def test_cholesky_inverse_errors_and_warnings(self, device, dtype):
+        # cholesky_inverse requires the input to be at least 2 dimensional tensor
+        a = torch.randn(2, device=device, dtype=dtype)
+        with self.assertRaisesRegex(RuntimeError, "must have at least 2 dimensions"):
+            torch.cholesky_inverse(a)
+
+        # cholesky_inverse requires a square matrix
+        a = torch.randn(2, 3, device=device, dtype=dtype)
+        with self.assertRaisesRegex(RuntimeError, "must be batches of square matrices"):
+            torch.cholesky_inverse(a)
+
+        # if non-empty out tensor with wrong shape is passed an error is thrown
```

Comment:
This test will need to be commented out (we should have the flag to make warnings that appear once always warn, and when that's available we can re-enable this test)

Reply:
I replaced this one with the test for warning.